### PR TITLE
pagination keyboard support

### PR DESCRIPTION
**Pull request by xujing-shen**
#### Please provide a link to the associated issue.
#89

### CODE REVIEW

File: src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -92,6 +90,16 @@ export class Pagination extends TsxComponent<Props> {
...
   }
 
+  private navigateToFirst() {
+    this.selectedPage = 1;
+    this.$emit('page-change', this.selectedPage);
+  }
+
+  private navigateToLast() {
+    this.selectedPage = this.numberOfPages;
+    this.$emit('page-change', this.selectedPage);
```
Comment (ChristianKienle) [inline src/components/Pagination/Pagination.tsx]: Have you considered calling the event 'update:selectedPage' in order to support the .sync-modifier?

https://vuejs.org/v2/guide/components-custom-events.html#sync-Modifier

File: src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -108,9 +116,36 @@ export class Pagination extends TsxComponent<Props> {
...
+    } else if (key === 'End') {
+      this.navigateToLast();
+    } else if (key === 'ArrowLeft' || key === 'ArrowUp' || key === 'PageUp') {
+      this.navigateBack();
+    } else if (key === 'ArrowRight' || key === 'ArrowDown' || key === 'PageDown') {
+      this.navigateForward();
+    }
+  }
+
+  public mounted() {
```
Comment (ChristianKienle) [inline src/components/Pagination/Pagination.tsx]: Have you considered doing it in the render method?

```
  public render() {
    return (
      <div
        class='fd-pagination'
        id='fd-pagination'
        on-keydown={this.keyHandler}
      >
…
```
Then the id attribute can go.

Comment (xujing-shen) [reply src/components/Pagination/Pagination.tsx]: Yes, on-keydown does the job of registering event listener. But how about deregistering when the component is destroyed. Do you have idea of how i can refer to the component in destroyed() without using id

**Comment by xujing-shen** [comment]
To make the keyboard navigation works, firstly need to use tab key or mouse to make the pagination component in focus, then the keyboard event is propagated and handled.

File: src/components/Pagination/Pagination.tsx (hunk 1 above)
Comment (xujing-shen) [reply src/components/Pagination/Pagination.tsx]: Done

**Pull request merged**